test/sandbox/test/rendering_test.rb

```diff
@@ -15,6 +15,17 @@ def test_render_in_view_context
     assert_selector("div", text: "hello,world!")
   end
 
+  def test_render_in_view_context_forwards_arguments
+    @foo = "foo"
+    @bar = "bar"
+
+    render_in_view_context(@foo, bar: @bar) do |foo, bar:|
+      render(MyComponent.new) { foo + bar }
```

Comment:
Not sure if it matters, but this test does not match the PR description, which had a reference to the ivar inside the `render` call 🤔

Reply:
The example in the PR description that uses an ivar inside `render_in_view_context` is the one that doesn't work. There's nothing we can do about ivar access (since the value of `self` is changed by `instance_exec`), so there's no way to make the example work as written. The idea behind this PR is to be able to pass data _stored_ in ivars (and elsewhere) into the block as arguments (locals). It's an imperfect workaround, but I don't think there's much else we can do.